quickjs.c

```diff
@@ -18826,7 +18861,8 @@ static __exception int next_token(JSParseState *s)
                 if (*p == '\n') {
                     s->line_num++;
                     s->got_lf = TRUE; /* considered as LF for ASI */
-                    p++;
+                    s->eol = p++;
```

Comment:
Any chance of simplifying the logic with got_lf now? I didn't look into it but there might be some overlap?

Reply:
Not easily, I think. I can go into detail but they mean/describe slightly different things.